src/libpgagroal/management.c

```diff
@@ -1698,15 +1798,163 @@ write_ssl(SSL* ssl, void* buf, size_t size)
+ * into header socket.
+ *
+ * @param header: the header to write sender application name and its version into
+ * @param command: the server application name
+ * @param offset: the offset at which info has to be written
+ * @return 0 on success
+ */
+static int
+write_info(char* buffer, int command, int offset)
+{
+   int exec_version= pgagroal_executable_version_number(PGAGROAL_VERSION, sizeof(PGAGROAL_VERSION));
+   if (exec_version == 1)
+   {
+      return 1;
+   }
+
+   char* exec_name = pgagroal_executable_name(command);
+   if (exec_name == NULL)
+   {
+      return 1;
+   }
+
+   struct pgagroal_version_info info = {
+      .command = command,
+      .version = exec_version,
+   };
+
+   memcpy(&(info.s), S, sizeof(info.s));
+   memcpy(&(info.v), V, sizeof(info.v));
+
+   memcpy(&(buffer[offset]), info.s, sizeof(info.s));
+   pgagroal_write_int32(&(buffer[offset + 2]), info.command);
+
+   memcpy(&(buffer[offset + 6]), info.v, sizeof(info.v));
+   pgagroal_write_int32(&(buffer[offset + 9]), info.version);
+
+   pgagroal_log_debug("%s version %d", exec_name, exec_version);
+
+   return 0;
+}
+
+/*
+ * Utility function to convert PGAGROAL_VERSION into a number.
+ */
+static int
+pgagroal_executable_version_number(char* version, size_t version_size)
+{
+   int major;
+   int minor;
+   int patch;
+
+   long val;
+
+   char* del = ".";
+   char* endptr;
+   
+   if (version == NULL)
+   {
+      version = PGAGROAL_VERSION;
+	  version_size = sizeof(version);
+   }
+
+   char buf[version_size];
+
+   memcpy(buf, version, sizeof(buf));
+   
+   char* token = strtok(buf, del);
+   val = strtol(token, &endptr, 10);
+   if (errno == ERANGE || val <= LONG_MIN || val >= LONG_MAX)
+   {
+      goto error;
+   }
+   major = (int)val;
+
+   token = strtok(NULL, del);
+   val = strtol(token, &endptr, 10);
+   if (errno == ERANGE || val <= LONG_MIN || val >= LONG_MAX)
+   {
+      goto error;
+   }
+   minor = (int)val;
+
+   token = strtok(NULL, del);
+   val = strtol(token, &endptr, 10);
+   if (errno == ERANGE || val <= LONG_MIN || val >= LONG_MAX)
+   {
+      goto error;
+   }
+   patch = (int)val;
+
+   int version_number = major * 100000 + minor * 1000 + 10 * patch;
+
+   if (version_number < INT_MIN || version_number > INT_MAX || version_number < 107000)
+   {
+      goto error;
+   }
+
+   return version_number;
+
+error:
+   pgagroal_log_debug("pgagroal_get_executable_number got overflowed value: %s %s", version, strerror(errno));
+   errno = 0;
+   return 1;
+}
+
+/*
+ * Utility function to convert back version_number into a string.
+ */
+static int
+pgagroal_executable_version_name(char* version_name, int version_number)
+{
+   version_name[0] = ((version_number / 100000) % 10) + '0';
+   version_name[1] = '.';
+   version_name[2] = ((version_number / 1000) % 10) + '0';
+   version_name[3] = '.';
+   version_name[4] = ((version_number / 10) % 10) + '0';
+   version_name[5] = '\0';
+
+   return 0;
+}
```

Comment:
I needed this function to add a string into the json output. I know It's kind of ugly.